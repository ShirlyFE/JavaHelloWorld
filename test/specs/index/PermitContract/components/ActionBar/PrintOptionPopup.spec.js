import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import PrintOptionPopup from 'index/PermitContract/components/ActionBar/PrintOptionPopup';

describe('index/PermitContract/components/ActionBar/PrintOptionPopup', () => {
  it('should render without errors', () => {
    const component = mount(<PrintOptionPopup />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should work fine', () => {
    const component = mount(<PrintOptionPopup />);

    expect(component).toBeTruthy();

    component.node.open(true);
    expect(component.find('div.is-open')).toHaveLength(1);
    expect(component.find('div.options-option-selected')).toHaveLength(1);

    const printOptions = component.find('div.options-option');
    expect(printOptions).toHaveLength(2);

    printOptions.at(0).simulate('click');

    expect(component.find('div.options-recurring')).toHaveLength(1);
    const recurringCheckbox = component.find('input[type="checkbox"]');
    recurringCheckbox.simulate('change', { target: {} });

    recurringCheckbox.node.value = true;
    recurringCheckbox.simulate('change', { target: { value: true } });

    printOptions.at(1).simulate('click');

    const printButton = component.find('button.btn-strong');
    printButton.simulate('click');

    const cancelButton = component.find('button.btn-secondary');
    cancelButton.simulate('click');

    component.node.open(false);
    expect(component.find('div.options-option')).toHaveLength(2);
  });

  it('handlePopupConfirm correctly', () => {
    const component = mount(<PrintOptionPopup />);
    const instance = component.instance();
    const pageContainer = document.createElement('div');
    pageContainer.classList.add('page-container');

    component.node.open(true);
    expect(component.find('div.is-open')).toHaveLength(1);
    component.setState({ containsRecurring: true})
    document.body.appendChild(pageContainer);
    instance.handlePopupConfirm();
    expect(component.find('div.is-open')).toHaveLength(0);
    expect(pageContainer.classList.contains('printing'));

    component.setState({ option: null });
    component.node.open(true);
    expect(component.find('div.is-open')).toHaveLength(1);

    instance.handlePopupConfirm();
    expect(component.find('div.is-open')).toHaveLength(0);
  })
});
