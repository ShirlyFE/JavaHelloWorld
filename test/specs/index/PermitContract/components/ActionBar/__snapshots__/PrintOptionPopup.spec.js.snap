// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/PermitContract/components/ActionBar/PrintOptionPopup should render without errors 1`] = `
<PrintOptionPopup>
  <Alert
    onCancel={[Function]}
    onClose={[Function]}
    onConfirm={[Function]}
  >
    <Modal
      className="modal--alert undefined"
      onClose={[Function]}
      shown={false}
    >
      <div
        className="modal-wrap"
        role="dialog"
      >
        <div
          className="modal modal--alert undefined"
          onKeyDown={[Function]}
        >
          <section
            className="modal-box"
          >
            <header
              className="modal-header"
            >
              <h3
                className="modal-title"
              />
              <span
                aria-label="close-dialog"
                className="icon-close modal-close"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex="0"
              />
            </header>
            <div
              className="modal-body"
            >
              <div
                className="options"
              >
                <div
                  className="options-option options-option-selected"
                  onClick={[Function]}
                >
                  <div />
                  <div>
                    <i
                      className="icon icon-check-circle"
                    />
                  </div>
                </div>
                <div
                  className="options-option "
                  onClick={[Function]}
                >
                  <div>
                    Schedule
                  </div>
                  <div />
                </div>
                <div
                  className="options-recurring"
                >
                  <Checkbox
                    checked={false}
                    className="options-recurring-checkbox"
                    onChange={[Function]}
                  >
                    <CheckboxComponent
                      checked={false}
                      className="options-recurring-checkbox"
                      onChange={[Function]}
                    >
                      <label
                        className="checkbox-wrapper options-recurring-checkbox"
                        htmlFor="checkbox_attr_for_1"
                      >
                        <span
                          className="checkbox"
                        >
                          <input
                            aria-checked={false}
                            checked={false}
                            className="checkbox__input"
                            id="checkbox_attr_for_1"
                            onChange={[Function]}
                            role="checkbox"
                            type="checkbox"
                          />
                          <span
                            className="checkbox__inner"
                          />
                        </span>
                      </label>
                    </CheckboxComponent>
                  </Checkbox>
                  Display all bookings for recurring
                </div>
              </div>
            </div>
            <div
              className="modal-footer"
            >
              <Button
                className=""
                loading={false}
                menuData={Array []}
                noSubmit={false}
                onClick={[Function]}
                type="secondary"
              >
                <button
                  className="btn btn-secondary"
                  menuData={Array []}
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  type="submit"
                >
                  <span>
                    Cancel
                  </span>
                </button>
              </Button>
              <Button
                className=""
                loading={false}
                menuData={Array []}
                noSubmit={false}
                onClick={[Function]}
                type="strong"
              >
                <button
                  className="btn btn-strong"
                  menuData={Array []}
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  type="submit"
                >
                  <span>
                    OK
                  </span>
                </button>
              </Button>
            </div>
          </section>
        </div>
        <div
          className="modal-mask"
        />
      </div>
    </Modal>
  </Alert>
</PrintOptionPopup>
`;
